eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -61,13 +134,285 @@ protected function configure()
         }
 
-        $this->searchIndexer->createSearchIndex($output, (int) $iterationCount, empty($noCommit));
+        if (!$this->searchIndexer instanceof IncrementalIndexer) {
+            $output->writeln(<<<EOT
+DEPRECATED:
+Running indexing against an Indexer that has not been updated to use IncrementalIndexer abstract.
+
+Options that won't be taken into account:
+- since
+- content-ids
+- processes
+- no-purge
+EOT
+            );
+            $this->searchIndexer->createSearchIndex($output, (int) $iterationCount, !$commit);
+        } else {
+            $output->writeln('Re-indexing started for search engine: ' . $this->searchIndexer->getName());
+            $output->writeln('');
+
+            $return = $this->indexIncrementally($input, $output, $iterationCount, $commit);
+
+            $output->writeln('');
+            $output->writeln('Finished re-indexing');
+
+            return $return;
+        }
+    }
+
+    protected function indexIncrementally(InputInterface $input, OutputInterface $output, $iterationCount, $commit)
+    {
+        if ($contentIds = $input->getOption('content-ids')) {
+            $output->writeln('Indexing list of content id\'s');
+
+            return $this->searchIndexer->updateSearchIndex(explode(',', $contentIds), $commit);
+        }
+
+        if ($since = $input->getOption('since')) {
+            $stmt = $this->getStatementContentSince(new DateTime($since));
+            $count = (int)$this->getStatementContentSince(new DateTime($since), true)->fetchColumn();
+            $purge = false;
+        } elseif ($locationId = (int) $input->getOption('subtree')) {
+            $stmt = $this->getStatementSubtree($locationId);
+            $count = (int) $this->getStatementSubtree($locationId, true)->fetchColumn();
+            $purge = false;
+        } else {
+            $stmt = $this->getStatementContentAll();
+            $count = (int) $this->getStatementContentAll(true)->fetchColumn();
+            $purge = !$input->getOption('no-purge');
+        }
+
+        if (!$count) {
+            $output->writeln('<error>Could not find any items to index, aborting.</error>');
+
+            return 1;
+        }
+
+        $iterations = ceil($count / $iterationCount);
+        $processCount = (int) $input->getOption('processes');
+        $processCount = $processCount > $iterations ? $iterations : $processCount;
+        $processMessage = $processCount > 1 ? "using $processCount parallel processes" : 'using single process';
+
+        if ($purge) {
+            $output->writeln('Purging index...');
+            $this->searchIndexer->purge();
+
+            $output->writeln(
+                "<info>Re-Creating index for {$count} items across $iterations iteration(s), $processMessage:"
+            );
+        } else {
+            $output->writeln(
+                "<info>Refreshing index for {$count} items across $iterations iteration(s), $processMessage:"
```

Comment:
I managed to run it for Solr and noticed you have unclosed `<info>` tag here and above which makes progress bar green as well, white looks better ;)

Reply:
fixed in d7219ab